tests/gtests/internals/test_sdpa.cpp

```diff
@@ -0,0 +1,1817 @@
+/*******************************************************************************
+* Copyright 2025 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <dnnl_test_common.hpp>
+#include <gtest/gtest.h>
+
+#include "sdpa_internal.hpp"
+
+#include <oneapi/dnnl/dnnl.hpp>
+
+#include <memory>
+#include <random>
+
+using std::vector;
+
+using mdt = memory::data_type;
+
+enum class quantize_type {
```

Comment:
clang-tidy will complaint on a type doesn't have `_t` suffix. Same for `mask_type`.

Reply:
I think clang-tidy will only complain about actual classes and not enum classes. The clang-tidy tests seem to pass in the lightweight checks.